storm-core/src/jvm/org/apache/storm/utils/ConfigUtils.java

```diff
@@ -0,0 +1,711 @@
+        if (mockedSupervisorStormDistRoot != null) {
+            return null;
+        }
+        return stormDistPath(supervisorLocalDir(conf)); // TODO: no need to forceMake here?, clj does not.
+    }
+
+    public static String supervisorStormDistRoot(Map conf, String stormId) throws IOException {
+        if (mockedSupervisorStormDistRoot != null) {
+            return null;
+        }
+        return supervisorStormDistRoot(conf) + FILE_SEPARATOR + stormId; // TODO: need to (url-encode storm-id)? Not.
+    }
+
+    public static String supervisorStormJarPath(String stormRoot) {
+        String ret = "";
+        // we do this since to concat a null String will actually concat a "null", which is not the expected: ""
+        if (stormRoot != null) {
+            ret = stormRoot;
+        }
+        return (ret + FILE_SEPARATOR + "stormjar.jar");
+    }
+
+    /* Never get used TODO : may delete it*/
+    public static String supervisorStormMetaFilePath(String stormRoot) {
+        String ret = "";
+        if (stormRoot != null) {
+            ret = stormRoot;
+        }
+        return (ret + FILE_SEPARATOR + "storm-code-distributor.meta");
+    }
+
+    public static String supervisorStormCodePath(String stormRoot) {
+        String ret = "";
+        if (stormRoot != null) {
+            ret = stormRoot;
+        }
+        return (ret + FILE_SEPARATOR + "stormcode.ser");
+    }
+
+    public static String supervisorStormConfPath(String stormRoot) {
+        String ret = "";
+        if (stormRoot != null) {
+            ret = stormRoot;
+        }
+        return (ret + FILE_SEPARATOR + "stormconf.ser");
+    }
+
+    public static String supervisorTmpDir(Map conf) throws IOException {
+        String ret = supervisorLocalDir(conf) + FILE_SEPARATOR + "tmp";
+        FileUtils.forceMkdir(new File(ret));
+        return ret;
+    }
+
+    public static String supervisorStormResourcesPath(String stormRoot) {
+        String ret = "";
+        // we do this since to concat a null String will actually concat a "null", which is not the expected: ""
+        if (stormRoot != null) {
+            ret = stormRoot;
+        }
+        return (ret + FILE_SEPARATOR + RESOURCES_SUBDIR);
+    }
+
+    //For testing only
+    // for java
+    // try (SetMockedSupervisorState mocked = new SetMockedSupervisorState(conf)) {
+    //    run test ...
+    // }
+    //
+    // for clojure
+    // (with-open [mock (SetMockedSupervisorState. conf)]
+    //     run test ...)
+    public static class SetMockedSupervisorState implements Closeable {
+        public SetMockedSupervisorState(Map conf) {
+            mockedSupervisorState = conf;
+        }
+        @Override
+        public void close() {
+            mockedSupervisorState = null;
+        }
+    }
+    private static Map mockedSupervisorState = null;
+    public static LocalState supervisorState(Map conf) throws IOException {
+        if (mockedSupervisorState != null) {
+            return null;
```

Comment:
Actually here we should return null. I know it is not straightforward: by default, mockedSupervisorState is null. Only when the test set this variable to be not null then we return "null".